packages/dds/tree/docs/wip/schema-evolution.md

```diff
@@ -0,0 +1,583 @@
+This means that a staged rollout might not be necessary.
+Each kind of schema change enumerated in the [The Evolution of Schema](#the-evolution-of-schema) section above will be examined below to see whether it requires a staged rollout to accomplish or whether there's a **shortcut** to do it without.
+
+This analysis is also summarized in a succinct table in [Appendix A](#appendix-a-shortcut-matrix).
+
+1. **Add an allowed type.**
+   * We can provide a special "catch-all" type that will be used for any type that is unrecognized.
+     For example, imagine a digital whiteboard app that renders various types of content to the screen.
+     The developer chooses "Ink", "Sticky Note" and "Unknown" (the catch-all) as the possible node types allowed in the whiteboard canvas field.
+     To render an "Unknown", the app makes a gray box that says "This content is not supported, please update to the latest version of the application", but the app otherwise is fully functional as it has been (it is _not_ an error).
+     In a future version of the application, the possible node types are increased to "Ink", "Sticky Note", "Unknown" and "Image".
+     A version of the app with the original view schema will interpret Images as "Unknown" and render the gray box.
+     Therefore the new version, with the new view schema, can go ahead and upgrade the stored schema as well without worrying about doing the staged rollout process.
+
+     Clearly, when this strategy makes sense for the application, it immensely reduces the amount of work required to add a new node type.
+     However, it must be employed preemptively.
+     The developer needs to include the "Unknown" type ahead of time, anticipating that "there might be some new stuff here someday".
+   * To mitigate that problem, another option would be to have the "Unknown" type be automatic - or "opt-out" rather than "opt-in".
+     That is, all applications must handle the "Unknown" type in every field, unless they intentionally make the decision not to.
+   * A third approach would be to automatically filter out unrecognized node types where applicable.
+     In the example above, the original client would simply not be able to observe that there are Image nodes in the canvas field - the field would appear shorter than it does to the upgraded clients if there are images in it.
+     This might be appropriate for sequence fields, but would not work for required fields because a required field must not appear empty.
+
+     > TODO: This is one of Noah's off-the-cuff ideas. Whether or not filtering makes sense for optional fields (or at all) is debatable.
+
+     This strategy could be used in combination with the "Unknown" type - if Unknown is in the field's set of allowed types, then the foreign type will present as an Unknown, otherwise it will be filtered.
```

Comment:
I'd say - it's similar to above, as in - it's hard to ensure that unknown content roundtrips correctly.
If the app is not testing it (doing end-to-end coverage using some artificial scenarios), then it's not working, whether runtime provides this support or not. And if they are testing it, they likely know where they need to handle unknown content (it needs to be explicit), so option # 1 is sufficient.